common/eth2/src/types.rs

```diff
@@ -1008,15 +1009,40 @@ impl FromStr for Accept {
     type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "application/octet-stream" => Ok(Accept::Ssz),
-            "application/json" => Ok(Accept::Json),
-            "*/*" => Ok(Accept::Any),
-            _ => Err("accept header cannot be parsed.".to_string()),
+        let mimes = parse_accept(s);
+
+        // [q-factor weighting]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
+        let mut highest_mime: Option<Mime> = None;
```

Comment:
In practice grabbing the highest MIME is probably fine, but we could be slightly more robust against strange requests by sorting the MIMEs by their q-factor and then iterating until we find one we understand (JSON, octet stream, `*/*`).